proto/buf/alpha/registry/v1alpha1/resolve.proto

```diff
@@ -107,3 +113,30 @@ message GetLocalModulePinsResponse {
   // This includes the transitive deps.
   repeated buf.alpha.module.v1alpha1.ModulePin dependencies = 2;
 }
+
+message ResolveReferenceRequest {
+  // The plugin reference to resolve.
+  ResolveReferencePlugin plugin_reference = 1;
+  // The module reference to resolve.
+  LocalModuleReference module_reference = 2;
+}
+
+// ResolveReferencePlugin is a plugin reference.
+// If the version is empty, this is a reference to the latest version.
+message ResolveReferencePlugin {
+  // The owner of the plugin.
+  // example: bufbuild
+  string owner = 1;
+  // The name of the plugin.
+  // example: connect-go
+  string name = 2;
+  // The version of the plugin.
+  // If empty, this is a reference to the latest version.
+  // example: v1.0.0
+  string version = 3;
```

Comment:
we also have plugin revisions - I could see either:

1. always using the latest revision (seems reasonable as we only do this generally to fix bugs; there's not much reason to use an older revision)
2. add a separate field for it (could be a `uint32`)
3. include it in the `version` string

what do you think? I think 1 is fine, but if so we might want to document it either on the message or on version.

Reply:
There have been some issues with old revisions of plugins being broken and old versions being requested, i feel like always getting the latest and documenting is a good idea because it means that there'll be less cached versions everywhere and we have the freedom to update revisions without any client side changes. 

if this ends up being the wrong decision we can add another field for it and it wouldn't be a breaking change

Reply:
agreed, sounds good to me.